Source/Extensions/Blazorise.LottieAnimation/wwwroot/lottie-animation.js

```diff
@@ -0,0 +1,80 @@
+import './vendors/lottie.js?v=1.1.2.0';
+
+import { getRequiredElement } from "../Blazorise/utilities.js?v=1.1.2.0";
+
+/**
+ * Initializes a new animation instance
+ * @param dotNetAdapter Reference to the dotnet object
+ * @param element Container element reference
+ * @param elementId Container element Id
+ * @param options Animation configuration options
+ * @returns Reference to the Lottie Animation
+ */
+export function initializeAnimation(dotNetAdapter, element, elementId, options) {
+    element = getRequiredElement(element, elementId);
+
+    if (!element) {
+        return;
+    }
+
+    options.container = element;
+
+    const animation = lottie.loadAnimation(options);
+    animation.setDirection(options.direction);
+    animation.setSpeed(options.speed);
+
+    // Add a function to set a flag for whether or not we're sending current frame updates
+    animation.setSendCurrentFrame = function (shouldSend) {
+        this.sendCurrentFrame = shouldSend;
+    }
+
+    // Add a function to set the loop property
+    animation.setLoop = function (loop) {
+        this.loop = loop;
+    }
+
+    animation.setSendCurrentFrame(options.sendCurrentFrame);
+    registerEvents(dotNetAdapter, animation);
+
+    return animation;
+}
+
```

Comment:
Can you also add a `destroy` method that will destroy the animation and remove any event listeners?

Reply:
There's already a `destroy` method the animations themselves that we call directly by returning a reference to the animation. 

That's what gets called in the [DisposeAnimation](https://github.com/Megabit/Blazorise/blob/905bb7b21526a85a5858b0cc9c5d537104eea937/Source/Extensions/Blazorise.LottieAnimation/LottieAnimation.razor.cs#L142) method, which is part of the component's `DisposeAsync` method.

Does that cover what we need?

Reply:
I've just now seen that you return `animation` object from JS and then store it in IJSObjectReference. We usually `export function destroy` and then call that one instead. And in IJSObjectReference is stored the JS module. I think it is better to stick to the same principle.

Reply:
Now that I think, you can leave it as it is. It's a good approach, making it easier to work with the `animation` object and call its methods.